ironfish/src/mining/poolShares.ts

```diff
@@ -206,13 +206,16 @@ export class MiningPoolShares {
     }
   }
 
-  async shareRate(): Promise<number> {
-    return (await this.recentShareCount()) / this.recentShareCutoff
+  async shareRate(publicAddress?: string): Promise<number> {
+    return (await this.recentShareCount(publicAddress)) / this.recentShareCutoff
   }
 
-  private async recentShareCount(): Promise<number> {
+  private async recentShareCount(publicAddress?: string): Promise<number> {
     const timestamp = Math.floor(new Date().getTime() / 1000) - this.recentShareCutoff
 
+    if (publicAddress) {
+      return await this.db.addressShareCountSince(publicAddress, timestamp)
+    }
     return await this.db.shareCountSince(timestamp)
```

Comment:
nit: We're mixing two patterns here, which is fine but it's just interesting. All these functions were parameterized with an optional arg, but all the DB functions were duplicated.